projects/packages/my-jetpack/_inc/components/product-cards-section/protect-card/scan-threats.tsx

```diff
@@ -0,0 +1,249 @@
+		const pluginsThreats = num_plugins_threats
+			? fromScanPlugins.reduce( ( accum, plugin ) => accum.concat( plugin.threats ), [] )
+			: [];
+		const themesThreats = num_themes_threats
+			? fromScanThemes.reduce( ( accum, theme ) => accum.concat( theme.threats ), [] )
+			: [];
+		const allThreats = [
+			...pluginsThreats,
+			...themesThreats,
+			...( core?.threats ?? [] ),
+			...database,
+			...files,
+		];
+		return allThreats.reduce(
+			( accum, threat ) => ( threat.severity >= 5 ? ( accum += 1 ) : accum ),
+			0
+		);
+	}, [ fromScanPlugins, fromScanThemes, scanData ] );
+
+	if ( isPluginActive && isSiteConnected ) {
+		if ( hasProtectPaidPlan ) {
+			if ( numThreats ) {
+				return (
+					<ThreatStatus
+						numThreats={ numThreats }
+						criticalThreatCount={ criticalScanThreatCount }
+						tooltipContent={ scanThreatsTooltip }
+					/>
+				);
+			}
+			return <ScanStatus status={ 'success' } tooltipContent={ scanThreatsTooltip } />;
+		}
+		return numThreats ? (
+			<ThreatStatus numThreats={ numThreats } tooltipContent={ scanThreatsTooltip } />
+		) : (
+			<ScanStatus status={ 'partial' } tooltipContent={ scanThreatsTooltip } />
+		);
+	}
+
+	return <ScanStatus status={ 'off' } tooltipContent={ scanThreatsTooltip } />;
+};
+
+/**
+ * ThreatStatus component
+ *
+ * @param {PropsWithChildren} props - The component props
+ * @param {number} props.numThreats - The number of threats
+ * @param {number} props.criticalThreatCount - The number of critical threats
+ * @param {TooltipContent[ 'scanThreatsTooltip' ]} props.tooltipContent - The number of critical threats
+ * @returns {ReactElement} rendered component
+ */
+function ThreatStatus( {
+	numThreats,
+	criticalThreatCount,
+	tooltipContent,
+}: {
+	numThreats: number;
+	criticalThreatCount?: number;
+	tooltipContent?: TooltipContent[ 'scanThreatsTooltip' ];
+} ) {
+	const { recordEvent } = useAnalytics();
+	const useTooltipRef = useRef< HTMLButtonElement >();
+	const isMobileViewport: boolean = useViewportMatch( 'medium', '<' );
+	const [ isPopoverVisible, setIsPopoverVisible ] = useState( false );
+
+	const toggleTooltip = useCallback(
+		() =>
+			setIsPopoverVisible( prevState => {
+				if ( ! prevState === true ) {
+					recordEvent( 'jetpack_protect_card_tooltip_open', {
+						page: 'my-jetpack',
+						feature: 'jetpack-protect',
+						location: 'scan',
+						status: 'alert',
+						hasPaidPlan: true,
+						threats: numThreats,
+					} );
+				}
+				return ! prevState;
+			} ),
+		[ numThreats, recordEvent ]
+	);
+	const hideTooltip = useCallback( () => {
+		// Don't hide the tooltip here if it's the tooltip button that was clicked (the button
+		// becoming the document's activeElement). Instead let toggleTooltip() handle the closing.
+		if ( useTooltipRef.current && ! useTooltipRef.current.contains( document.activeElement ) ) {
+			setIsPopoverVisible( false );
+		}
+	}, [ setIsPopoverVisible, useTooltipRef ] );
+
+	return criticalThreatCount ? (
```

Comment:
Can we still split it into two returns?

Like:

if ( criticalThreatCount ) {
   return <>Errors</>
}

return <>Default</>

Currently, it's easy to lose `:` operator in between these two layouts